### ISSUE TITLE
Suggestions on best way to integrate user acceptance testing into our workflow?

### ISSUE THREAD

**Issue by eddiethefoot** [ZenHubIO/support#717]
I'd love any suggestions on how other community members have integrated user acceptance testing into their workflow.

Generally after we get a pull request that is merged, we have a release that's automatically pushed to our QA environment. From here, a QA team will test for usability and to make sure the designs have been implemented correctly. I believe that currently if a PR is merged, the ticket is automatically closed and placed in the Closed column? 
Preferably we'd like a QA column and then to move things to DONE only after we've confirmed that changes are live in our production environment. I'm looking for suggestions on how to best achieve this.

Love the product so far!

**Comment by azenMatt** [comment]
Hi @eddiethefoot - thanks for the kind words :) 

The work flow you have described is how we are currently working as well. We have some changes coming out in the next 1-2 weeks that should really help you here. In particular the ability to have PR's follow Issue cards (designs are here - https://github.com/ZenHubIO/support/issues/466 ) 

This should make it much easier to show a merged PR next to an open issue - allowing the QA team to understand what code has been changed as part of the issue. 

Open to ideas on how to help improve QA flows. 



> I believe that currently if a PR is merged, the ticket is automatically closed and placed in the Closed column? 
> Preferably we'd like a QA column and then to move things to DONE only after we've confirmed that changes are live in our production environment

For this - are you referring to the PR or the Issue card?

**Comment by eddiethefoot** [comment]
Hey Matt,

Thanks for the response! 

> For this - are you referring to the PR or the Issue card?

I was referring to the Issue Card, but I may have been mistaken. 

After some internal discussion, we've decided to migrate fully to ZenHub. The workflow we've settled on is as follows:

**New Issues** >> **Backlog** >> **To Do** >> **In Progress** >> **Ready for QA** >> **Ready for Release** >> Closed (Done)

We've settled that when a PR is merged, we'd call it Ready for QA (UAT or automated), and then QA would mark as Ready for Release, and then once released, we'd close the tickets manually.

**Comment by azenMatt** [comment]
That's great news :) 



> I was referring to the Issue Card, but I may have been mistaken.

The reason I mentioned this is because I find that some teams (including ours!) get caught out with the GitHub keywords - https://help.github.com/articles/closing-issues-via-commit-messages/ 

So if a PR makes a comment like `closes #456` when that PR is merged, the issue referenced gets automatically closed. This is great for some workflows, but we found it caused confusion when issues were closed as the PR was merged into our test environment. 

#466 should make this much easier by allowing PRs and Issues to be linked without forcing closure on merge 

Please keep me updated if you have any other feedback on improving test-flow in ZenHub + GitHub!

**Comment by eddiethefoot** [comment]
Thanks! Yeah that's exactly what we started doing as well. We're referencing the issue without closing it via comments. 👍
